### PR TITLE
refactor: limit complexity to increase maintainability

### PR DESCRIPTION
**Pull request by nicoandrewss**

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -2,179 +2,401 @@ package gorabbitmq
...
 
-	reconnectChan        chan struct{}
-	reconnectFailChanMtx *sync.Mutex
-	reconnectFailChan    chan error
+	startRecoveryChan  chan struct{}
+	recoveryFailedChan chan error
+
+	consumerRecoveryChan chan error
+
+	runningConsumers int
```
Comment (MaxBreida) [inline connection.go:33]: should this maybe be a atomic.Int32? 😄 
could this in any way be racy?

Comment (nicoandrewss) [reply connection.go:33]: no, it's never accessed concurrently

File: connection.go [hunk 2]
```diff
@@ -2,179 +2,401 @@ package gorabbitmq
...
+			net.JoinHostPort(
+				url.QueryEscape(settings.Host),
+				strconv.Itoa(settings.Port),
+			),
+		),
+	)
+
+	for i := 0; i < len(options); i++ {
+		options[i](opt)
+	}
```
Comment (MaxBreida) [inline connection.go:59]: btw why not a simple range loop? 😄

**Review by MaxBreida — APPROVED** [review]
rest seems good to me

**Pull request merged**